### ISSUE TITLE
Serving using ExportStrategy

### ISSUE THREAD

**Issue by dennybritz** [google/seq2seq#4]
Should figure out how to export models for serving, I think Tensorflow does provide something like an `ExportStrategy` that can be passed to the estimator and it will occasionally export the model.

**Comment by gidim** [comment]
It might be easier to add support for using `feed_dict` as an alternative for `InputPipeline`?
Many TF models use that paradigm rather than reading the input data directly from the graph.

**Comment by skyw** [comment]
I think the standard way is freeze the model, then use feed_dict to feed input and get the output tensor.

**Comment by gidim** [comment]
@skyw that's correct but since the models in this repo use Readers you first have to declare a new input tensor to the model (Placeholder).

**Comment by xueyouluo** [comment]
Any updates of this feature? I agree with @gidim , currently the model depend of tf.Slim.Dataset which is hard to be used for online inference.

**Comment by amirj** [comment]
I think there is two options:

1. Load the model, add an input placeholder to the graph, remove the input pipeline from the graph and freeze it. As mentioned by @skyw 

2. [better option] leverage TFserving component to plug a model contains input pipeline with queues to a web server.

I think, the serving feature is very important feature for this project referred in other issues (#114).

**Comment by gidim** [comment]
@amirj does tfserving supports option two? If so could link to the documentation/code ? I couldn't find anything about it.

**Comment by amirj** [comment]
Input pipelines are [the standard way](https://www.tensorflow.org/performance/performance_guide#utilize_queues_for_reading_data) to feeding TF models. So, I think, its indeed possible. But the current documentation in TF serving module is not clear.

**Comment by gidim** [comment]
@amirj - the standard way when performing training. for inference most models use feed_dict.
Usually it's easy to rebuild the graph with two placeholders but the code here is a a little complex and has quite a few dependencies which makes it harder.

**Comment by amirj** [comment]
It's strange if TF serving is not compatible with input queues.
It's not a good design to leverage different feeding in the training and inferencing. For example, one application of ExportStrategy is online inference during the training phase #114 . Rebuilding the graph in this situation is not good.

**Comment by sathyarr** [comment]
@amirj @gidim

do you think the approach followed [here](https://stackoverflow.com/questions/52120188/how-to-get-tensorflow-served-model-to-pull-from-passed-in-input-and-not-local-ba) to serve the graph is recommended?

**Comment by sathyarr** [comment]
> Should figure out how to export models for serving, I think Tensorflow does provide something like an `ExportStrategy` that can be passed to the estimator and it will occasionally export the model.

@dennybritz 
I have managed to export through `export_strategies` in `Experiment`.
But, facing problem with preprocessing and embedding.
Kindly have a look [here](https://github.com/tensorflow/tensorflow/pull/12759#issuecomment-467755590) and [here](https://stackoverflow.com/questions/54899188/error-model-att-seq2seq-minimum0-is-both-fed-and-fetched)

Are things performed in _preprocess exported to Graph?
